src/peft/tuners/tuners_utils.py

```diff
@@ -486,30 +484,43 @@ def check_target_module_exists(config, key: str) -> bool | re.Match[str] | None:
         None if no match found
     """
     if isinstance(config.target_modules, str):
-        target_module_found = re.fullmatch(config.target_modules, key)
+        target_module_found = (config.target_modules == key) or re.fullmatch(config.target_modules, key)
+    elif key in config.target_modules:
+        # this module is specified directly in target_modules
+        target_module_found = True
     else:
-        target_module_found = key in config.target_modules or any(
-            key.endswith(f".{target_key}") for target_key in config.target_modules
-        )
-        is_using_layer_indexes = getattr(config, "layers_to_transform", None) is not None
-        layer_indexing_pattern = getattr(config, "layers_pattern", None)
+        target_module_found = any(key.endswith(f".{target_key}") for target_key in config.target_modules)
 
+        layer_indexes = getattr(config, "layers_to_transform", None)
+        layers_pattern = getattr(config, "layers_pattern", None)
+
+        is_using_layer_indexes = layer_indexes is not None and (
+            len(layer_indexes) != 0 if isinstance(layer_indexes, list) else True
+        )
         if is_using_layer_indexes and target_module_found:
-            layers_pattern = COMMON_LAYERS_PATTERN if layer_indexing_pattern is None else layer_indexing_pattern
-            layers_pattern = [layers_pattern] if isinstance(layers_pattern, str) else layers_pattern
-
-            for pattern in layers_pattern:
-                layer_index = re.match(f".*.{pattern}\.(\d+)\.*", key)
-                if layer_index is not None:
-                    layer_index = int(layer_index.group(1))
-                    if isinstance(config.layers_to_transform, int):
-                        target_module_found = layer_index == config.layers_to_transform
-                    else:
-                        target_module_found = layer_index in config.layers_to_transform
-
-                    break
+            # to keep up with the previous results(if layers_pattern is an empty list, ret is True)
```

Comment:
This whole comment is not clear to me, could you please try to word it differently? Note that the comment should make sense as is, even if the reader is not aware of how the code looked like previously.

Reply:
Sorry for that, I explain it above, and it will be changed later